syntax = "proto3";

package user;

service UserService {
  rpc RegisterUser (RegisterRequest) returns (UserResponse);
  rpc LoginUser (LoginRequest) returns (UserResponse);
  rpc GetUser (GetUserRequest) returns (UserResponse);
}

message RegisterRequest {
  string username = 1;
  string email = 2;
  string password = 3;
}

message LoginRequest {
  string email = 1;
  string password = 2;
}

message GetUserRequest {
  string user_id = 1;
}

message UserResponse {
  string user_id = 1;
  string username = 2;
  string email = 3;
  string created_at = 4;
}